File: contest-management-client-server-java/ContestMgmtNetworking/src/main/java/contestmgmt/networking/dto/RegistrationDTO.java
package contestmgmt.networking.dto;

import java.io.Serializable;

public class RegistrationDTO implements Serializable {
    private Long participantId;
    private String firstName;
    private String lastName;
    private int age;

    private Long competitionId;
    private String competitionType;
    private String ageCategory;

    public Long getParticipantId() {
        return participantId;
    }

    public void setParticipantId(Long participantId) {
        this.participantId = participantId;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public Long getCompetitionId() {
        return competitionId;
    }

    public void setCompetitionId(Long competitionId) {
        this.competitionId = competitionId;
    }

    public String getCompetitionType() {
        return competitionType;
    }

    public void setCompetitionType(String competitionType) {
        this.competitionType = competitionType;
    }

    public String getAgeCategory() {
        return ageCategory;
    }

    public void setAgeCategory(String ageCategory) {
        this.ageCategory = ageCategory;
    }

    @Override
    public String toString() {
        return "RegistrationDTO { participantId: %s, firstName: %s, lastName: %s, age: %d, competitionId: %s, competitionType: %s, ageCategory: %s }"
                .formatted(participantId.toString(), firstName, lastName, age, competitionId.toString(), competitionType, ageCategory);
    }
}

File: contest-management-client-server-cs/networking/dto/RegistrationDTO.cs
#nullable disable
using System;
using model;

namespace networking.dto
{
    [Serializable]
    public class RegistrationDTO
    {
        public long ParticipantId { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public int Age { get; set; }

        public long CompetitionId { get; set; }

        public string CompetitionType { get; set; }

        public string AgeCategory { get; set; }

        public static explicit operator Registration(RegistrationDTO registrationDto)
        {
            return new
            (
                new Participant
                {
                    Id = registrationDto.ParticipantId,
                    FirstName = registrationDto.FirstName,
                    LastName = registrationDto.LastName,
                    Age = registrationDto.Age
                },
                new Competition
                {
                    Id = registrationDto.CompetitionId,
                    CompetitionType = registrationDto.CompetitionType,
                    AgeCategory = registrationDto.AgeCategory
                }
            );
        }

        public static explicit operator RegistrationDTO(Registration registration)
        {
            var participant = registration.Participant;
            var competition = registration.Competition;
            return new RegistrationDTO
            {
                ParticipantId = participant.Id,
                FirstName = participant.FirstName,
                LastName = participant.LastName,
                Age = (int) participant.Age!,
                CompetitionId = competition.Id,
                CompetitionType = competition.CompetitionType,
                AgeCategory = competition.AgeCategory
            };
        }
    }
}